src/App.js

```diff
@@ -1,26 +1,36 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import data from './api/movies.json';
 
-export class App extends Component {
-  state = {
-    movies: data,
-  };
+export const App = () => {
+  const [movies, setMovies] = useState(data);
+
+  const addMovie = (movie) => {
+    if (!movie) {
+      return;
+    }
 
-  render() {
-    const { movies } = this.state;
+    const duplicate = movies
+      .some(film => film.imdbId === movie.imdbId);
```

Comment:
You may not create the duplicate. Just chek `if (movies.some(film => film.imdbId === movie.imdbId)` then do nothing. Else add new movie